pkg/asset/manifests/nutanix/cloudproviderconfig.go

```diff
@@ -0,0 +1,104 @@
+package nutanix
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	nutanixtypes "github.com/openshift/installer/pkg/types/nutanix"
+)
+
+// Config of Nutanix cloud provider
+// ref: https://github.com/nutanix-cloud-native/cloud-provider-nutanix/blob/main/pkg/provider/config/config.go
+type NutanixCloudConfig struct {
+	PrismCentral         NutanixPrismEndpoint `json:"prismCentral"`
+	TopologyDiscovery    TopologyDiscovery    `json:"topologyDiscovery"`
+	EnableCustomLabeling bool                 `json:"enableCustomLabeling"`
+}
+
+type TopologyDiscovery struct {
+	// Default type will be set to Prism via the newConfig function
+	Type               TopologyDiscoveryType `json:"type"`
+	TopologyCategories *TopologyCategories   `json:"topologyCategories"`
+}
+
+type TopologyDiscoveryType string
+
+const (
+	PrismTopologyDiscoveryType      = TopologyDiscoveryType("Prism")
+	CategoriesTopologyDiscoveryType = TopologyDiscoveryType("Categories")
+)
+
+type TopologyInfo struct {
+	Zone   string `json:"zone"`
+	Region string `json:"region"`
+}
+
+type TopologyCategories struct {
+	ZoneCategory   string `json:"zoneCategory"`
+	RegionCategory string `json:"regionCategory"`
+}
+
+type NutanixPrismEndpoint struct {
+	// address is the endpoint address (DNS name or IP address) of the Nutanix Prism Central or Element (cluster)
+	Address string `json:"address"`
+
+	// port is the port number to access the Nutanix Prism Central or Element (cluster)
+	Port int32 `json:"port"`
+
+	// Pass credential information for the target Prism instance
+	// +optional
+	CredentialRef *NutanixCredentialReference `json:"credentialRef,omitempty"`
+}
+
+type NutanixCredentialKind string
```

Comment:
this should be `CredentialKind`,  it's type will still be scoped from the package, eg `nutanix.CredentialKind`

Reply:
Fixed